core/structs-checker/structs-checker-macro/src/lib.rs

```diff
@@ -91,29 +91,66 @@ mod helper {
         quote! { &[#(#variants),*] }
     }
 
+    fn extract_type_info(ty: &syn::Type) -> TokenStream2 {
+        match ty {
+            syn::Type::Path(type_path) => {
+                let type_name =
+                    quote::format_ident!("{}", type_path.path.segments.last().unwrap().ident);
+                let generic_params = &type_path.path.segments.last().unwrap().arguments;
```

Comment:
Note that generics do not necessarily need to be part of the last segment. Consider for example: `<Apple as Fruit<Round>>::AssocType`. Here `AssocType` in `impl Fruit<Round> for Apple` can be a `Vec<Round>` or any other instantiation of a generic type.

Do we need to handle it now? Most plausibly not really, but should probably FIXME here.